client/src/Components/Timer/Timer.jsx

```diff
@@ -1,9 +1,55 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useSocket } from 'socketio-hooks';
 
 import InfoBox from 'Components/InfoBox';
 
 const Timer = () => {
-  return <InfoBox title="TIMER" value="00:22:33" padding={0.25} />;
+  const [seconds, setSeconds] = useState(0);
+  const [minutes, setMinutes] = useState(0);
+  const [isActive, setIsActive] = useState(false);
+
+  const reset = () => {
+    setSeconds(0);
+    setMinutes(0);
+    setIsActive(false);
+  };
+
+  const pad = (number, size) => {
+    var s = String(number);
+    while (s.length < (size || 2)) {
+      s = '0' + s;
+    }
+    return s;
+  };
+
+  useEffect(() => {
+    let interval = null;
+    if (isActive) {
+      interval = setInterval(() => {
+        if (minutes === 59 && seconds === 59) {
+          setSeconds(0);
+          setMinutes(0);
+        } else if (seconds === 59) {
+          setSeconds(0);
+          setMinutes(minutes => minutes + 1);
+        } else setSeconds(seconds => seconds + 1);
+      }, 1000);
+    } else if (!isActive && seconds !== 0) {
+      clearInterval(interval);
+    }
+    return () => clearInterval(interval);
+  }, [isActive, seconds, minutes]);

```

Comment:
Mam wrażenie że tu robimy nowy Interwal za każdym razem kiedy zmienia się seconds albo minutes. Wiem że react bedzie robił problem jak to sobie po prostu usuniemy z tablicy dlatego pogrzebałem trochę i znalazłem ciekawy art: https://overreacted.io/making-setinterval-declarative-with-react-hooks/

Myślę że możemy to wykorzystać